### PR TITLE
[Digital Guardian] Initial Release of Digital Guardian

### PR DESCRIPTION
**Pull request by kcreddy**
## Proposed commit message
```
Initial Release of Digital Guardian.

New integration to pull events and alerts from the 
Digital Guardian Analytics & Reporting Cloud (ARC) API.
```

## Checklist

- [ ] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/main/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [ ] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [ ] I have verified that Kibana version constraints are current according to [guidelines](https://github.com/elastic/elastic-package/blob/master/docs/howto/stack_version_support.md#when-to-update-the-condition).

## How to test this PR locally

`cd packages/digital_guardian && eval "$(elastic-package stack shellinit)" && elastic-package test system --generate  -v --defer-cleanup 30m `
```
--- Test results for package: digital_guardian - START ---
╭──────────────────┬─────────────┬───────────┬───────────┬────────┬───────────────╮
│ PACKAGE          │ DATA STREAM │ TEST TYPE │ TEST NAME │ RESULT │  TIME ELAPSED │
├──────────────────┼─────────────┼───────────┼───────────┼────────┼───────────────┤
│ digital_guardian │ arc         │ system    │ default   │ PASS   │ 40.953647792s │
╰──────────────────┴─────────────┴───────────┴───────────┴────────┴───────────────╯
--- Test results for package: digital_guardian - END   ---
Done
```

`cd packages/digital_guardian && eval "$(elastic-package stack shellinit)" && elastic-package test pipeline --generate  -v`
```
--- Test results for package: digital_guardian - START ---
╭──────────────────┬─────────────┬───────────┬────────────────────────────────────────────┬────────┬──────────────╮
│ PACKAGE          │ DATA STREAM │ TEST TYPE │ TEST NAME                                  │ RESULT │ TIME ELAPSED │
├──────────────────┼─────────────┼───────────┼────────────────────────────────────────────┼────────┼──────────────┤
│ digital_guardian │ arc         │ pipeline  │ (ingest pipeline warnings test-dg-arc.log) │ PASS   │     477.58ms │
│ digital_guardian │ arc         │ pipeline  │ test-dg-arc.log                            │ PASS   │  48.267458ms │
╰──────────────────┴─────────────┴───────────┴────────────────────────────────────────────┴────────┴──────────────╯
--- Test results for package: digital_guardian - END   ---
Done
```

## Related issues

- Closes https://github.com/elastic/integrations/issues/8545

## Screenshots

Screenshot 2024-07-31 at 4 35 59 PM

arc_overview

### CODE REVIEW

**Comment by kcreddy** [comment]
@jamiehynds, @andrewkroh I created `Digital Guardian` integration with `ARC` as one of the datastreams instead of `Digital Guardian ARC` integration. With this approach, we can add events from other DG products underneath same integration instead of creating new integrations, one for each product. 

Let me know if you think the other way works better.

File: packages/digital_guardian/_dev/deploy/docker/files/config.yml [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+      Accept:
+        - "application/json"
+    responses:
+      - status_code: 200
+        body: |-
+          {{ minify_json `
+          {
+            "fields": [
+                {
+                    "array": false,
```
Comment (efd6) [inline packages/digital_guardian/_dev/deploy/docker/files/config.yml:23]: Consistent indentation?

File: packages/digital_guardian/data_stream/arc/agent/stream/cel.yml.hbs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  client.secret: {{client_secret}}
+  token_url: {{auth_server_url}}/as/token.oauth2
+  scopes: {{scope}}
+redact:
+  fields: ~
+program: |
+  state.with(
+    request("POST",
+      state.url + "/rest/1.0/export_profiles/{{export_profile}}/export_and_ack"
+    ).with({
```
Comment (efd6) [inline packages/digital_guardian/data_stream/arc/agent/stream/cel.yml.hbs:28]: Just confirming that this is intentionally a body-less POST request?

File: packages/digital_guardian/data_stream/arc/agent/stream/cel.yml.hbs [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+              "message": e.encode_json(),
+            }))
+          })
+        : 
+          {
+            "events": dyn({
+              "error": {
+                "message": 'Response has no fields or data'
+              }
+            })
```
Comment (efd6) [inline packages/digital_guardian/data_stream/arc/agent/stream/cel.yml.hbs:46]: Is this case error-worthy? Would it not be better to have an error-less `{"events": []}` and detect that with no error message in the ingest pipeline? Unless the body here is passed to ingest there is nothing that the user can do with this information.

File: packages/digital_guardian/data_stream/arc/agent/stream/cel.yml.hbs (hunk 2 above)
Comment (kcreddy) [reply packages/digital_guardian/data_stream/arc/agent/stream/cel.yml.hbs:28]: As per comment [here](https://github.com/elastic/integrations/issues/8545#issuecomment-2231835068) and similar call [here](https://github.com/demisto/content/blob/master/Packs/DigitalGuardian/Integrations/DigitalGuardian/DigitalGuardian.py#L299-L300), this API doesn't contain body.

File: packages/digital_guardian/data_stream/arc/agent/stream/cel.yml.hbs (hunk 3 above)
Comment (kcreddy) [reply packages/digital_guardian/data_stream/arc/agent/stream/cel.yml.hbs]: Makes sense 👍🏼 . Updated as suggested.

File: packages/digital_guardian/_dev/deploy/docker/files/config.yml (hunk 1 above)
Comment (kcreddy) [reply packages/digital_guardian/_dev/deploy/docker/files/config.yml]: Indented properly.

File: packages/digital_guardian/data_stream/arc/agent/stream/cel.yml.hbs [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        (has(body.fields) && has(body.data) ? 
+          body.fields.map(e, e.name).as(field_names, {
+            "events": body.data.map(d, zip(field_names, d).as(e, {
+              "message": e.encode_json(),
+            }))
+          })
+        : 
+          {
+            {"events": []}
+          }
```
Comment (efd6) [inline packages/digital_guardian/data_stream/arc/agent/stream/cel.yml.hbs:43]:
```suggestion
          {"events": []}
```
(too much nesting)

**Review by efd6 — APPROVED** [review]
:bird:

**Pull request merged**